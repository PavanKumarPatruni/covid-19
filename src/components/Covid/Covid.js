import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

import CovidStats from '../../atoms/CovidStats';
import StateTable from '../../atoms/StateTable';

const propTypes = {
  getCovidStateData: PropTypes.func.isRequired,
  stateData: PropTypes.object.isRequired,
};

const Covid = ({ getCovidStateData, stateData }) => {
  const [stateList, setStateList] = useState([]);

  useEffect(() => {
    getCovidStateData();
  }, []);

  useEffect(() => {
    if (stateData && stateData.statewise) {
      setStateList(stateData.statewise);
    }
  }, [stateData]);

  const getStateTableComponent = () => {
    return stateList && stateList.length > 0 ? (
      <StateTable stateList={stateList} />
    ) : null;
  };

  const getStateStatsComponent = () => {
    return stateList && stateList.length > 0 ? (
      <CovidStats stats={stateList[0]} />
    ) : null;
  };

  return (
    <div className="">
      <div className="flex flex-center flex-column">
        {getStateStatsComponent()}
        <div className="sm-full-width overflow-x-auto">
          {getStateTableComponent()}
        </div>
      </div>
    </div>
  );
};

Covid.propTypes = propTypes;

export default Covid;
