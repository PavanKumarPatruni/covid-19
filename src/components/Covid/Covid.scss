@import '../../styles/index.scss';

.covid-container {
  background-color: $background;

  &.normal-mode {
    background-color: $white;
  }
}
