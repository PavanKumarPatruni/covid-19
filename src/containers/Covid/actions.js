import {
  COVID_GET_STATE_DATA,
  COVID_GET_STATE_DATA_SUCCESS,
  COVID_GET_STATE_DATA_FAILURE,
  COVID_GET_STATE_DATA_RESET,
  COVID_GET_DISTRICT_DATA,
  COVID_GET_DISTRICT_DATA_SUCCESS,
  COVID_GET_DISTRICT_DATA_FAILURE,
  COVID_GET_DISTRICT_DATA_RESET,
} from './constants';

export const getCovidStateData = () => {
  return {
    type: COVID_GET_STATE_DATA,
  };
};

export const getCovidStateDataSuccess = data => {
  return {
    type: COVID_GET_STATE_DATA_SUCCESS,
    data,
  };
};

export const getCovidStateDataFailure = error => {
  return {
    type: COVID_GET_STATE_DATA_FAILURE,
    error,
  };
};

export const getCovidStateDataReset = () => {
  return {
    type: COVID_GET_STATE_DATA_RESET,
  };
};

export const getCovidStateDistrictData = () => {
  return {
    type: COVID_GET_DISTRICT_DATA,
  };
};

export const getCovidStateDistrictDataSuccess = data => {
  return {
    type: COVID_GET_DISTRICT_DATA_SUCCESS,
    data,
  };
};

export const getCovidStateDistrictDataFailure = error => {
  return {
    type: COVID_GET_DISTRICT_DATA_FAILURE,
    error,
  };
};

export const getCovidStateDistrictDataReset = () => {
  return {
    type: COVID_GET_DISTRICT_DATA_RESET,
  };
};
