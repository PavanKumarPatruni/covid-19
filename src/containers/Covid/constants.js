export const COVID_GET_STATE_DATA = 'COVID/STATE/GET_DATA';
export const COVID_GET_STATE_DATA_SUCCESS = 'COVID/STATE/GET_DATA_SUCCESS';
export const COVID_GET_STATE_DATA_FAILURE = 'COVID/STATE/GET_DATA_FAILURE';
export const COVID_GET_STATE_DATA_RESET = 'COVID/STATE/GET_DATA_RESET';
