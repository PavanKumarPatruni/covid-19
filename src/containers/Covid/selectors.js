import { createSelector } from 'reselect';

const getCovidStateData = state => {
  return state.covidStateWiseData;
};

export const selectCovidStateData = createSelector(
  getCovidStateData,
  data => data.data
);
