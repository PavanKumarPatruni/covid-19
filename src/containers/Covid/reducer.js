import {
  COVID_GET_STATE_DATA,
  COVID_GET_STATE_DATA_SUCCESS,
  COVID_GET_STATE_DATA_FAILURE,
  COVID_GET_STATE_DATA_RESET,
  COVID_GET_DISTRICT_DATA,
  COVID_GET_DISTRICT_DATA_SUCCESS,
  COVID_GET_DISTRICT_DATA_FAILURE,
  COVID_GET_DISTRICT_DATA_RESET,
} from './constants';

import { INITIAL_STATE } from '../../redux/constants';

const getCovidStateDataReducer = (state = INITIAL_STATE, action) => {
  const { type, data, error } = action;
  switch (type) {
    case COVID_GET_STATE_DATA:
      return {
        ...state,
        isFetching: true,
      };
    case COVID_GET_STATE_DATA_SUCCESS:
      return {
        ...state,
        isFetching: false,
        isServerError: false,
        data,
        error: {},
      };
    case COVID_GET_STATE_DATA_FAILURE:
      return {
        ...state,
        isFetching: false,
        isServerError: true,
        error,
        data: {},
      };
    case COVID_GET_STATE_DATA_RESET:
      return {
        ...state,
        isFetching: false,
        isServerError: false,
        error: {},
        data: {},
      };
    default:
      return state;
  }
};

const getCovidStateDistrctDataReducer = (state = INITIAL_STATE, action) => {
  const { type, data, error } = action;
  switch (type) {
    case COVID_GET_DISTRICT_DATA:
      return {
        ...state,
        isFetching: true,
      };
    case COVID_GET_DISTRICT_DATA_SUCCESS:
      return {
        ...state,
        isFetching: false,
        isServerError: false,
        data,
        error: {},
      };
    case COVID_GET_DISTRICT_DATA_FAILURE:
      return {
        ...state,
        isFetching: false,
        isServerError: true,
        error,
        data: {},
      };
    case COVID_GET_DISTRICT_DATA_RESET:
      return {
        ...state,
        isFetching: false,
        isServerError: false,
        error: {},
        data: {},
      };
    default:
      return state;
  }
};

export { getCovidStateDataReducer, getCovidStateDistrctDataReducer };
