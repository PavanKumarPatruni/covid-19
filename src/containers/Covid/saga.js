import { call, put, takeLatest, all } from 'redux-saga/effects';
import apis from '../../utils/apis';
import { fetchGetData } from '../../utils/fetch';

import { getCovidStateDataSuccess, getCovidStateDataFailure } from './actions';

import { COVID_GET_STATE_DATA } from './constants';

export function* loadCovidStateData() {
  try {
    const response = yield call(fetchGetData, { url: apis.dataAPI });

    if (response) {
      yield put(getCovidStateDataSuccess(response));
    } else {
      yield put(getCovidStateDataFailure({ error: 'error' }));
    }
  } catch (err) {
    yield put(getCovidStateDataFailure({ error: err }));
  }
}

export function* getCovidStateData() {
  yield all([takeLatest(COVID_GET_STATE_DATA, loadCovidStateData)]);
}
