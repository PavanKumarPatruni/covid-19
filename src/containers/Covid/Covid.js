import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import Covid from '../../components/Covid';

import { getCovidStateData } from './actions';
import { selectCovidStateData } from './selectors';

export const mapDispatchToProps = dispatch => {
  return {
    getCovidStateData: () => {
      dispatch(getCovidStateData());
    },
  };
};

const mapStateToProps = createStructuredSelector({
  stateData: selectCovidStateData,
});

export default connect(mapStateToProps, mapDispatchToProps)(Covid);
