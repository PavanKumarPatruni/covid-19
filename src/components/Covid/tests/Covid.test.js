import React from 'react';
import { shallow, mount } from 'enzyme';

import Covid from '../Covid';

describe('Covid component', () => {
  const props = {
    getCovidStateData: () => {},
    stateData: {
      cases_time_series: {},
      statewise: [
        {
          active: '44351',
          confirmed: '67700',
          deaths: '2215',
          deltaconfirmed: '524',
          deltadeaths: '2',
          deltarecovered: '160',
          lastupdatedtime: '11/05/2020 15:41:22',
          recovered: '21130',
          state: 'Total',
          statecode: 'TT',
          statenotes: '',
        },
        {
          active: '44351',
          confirmed: '67700',
          deaths: '2215',
          deltaconfirmed: '524',
          deltadeaths: '2',
          deltarecovered: '160',
          lastupdatedtime: '11/05/2020 15:41:22',
          recovered: '21130',
          state: 'Total',
          statecode: 'AP',
          statenotes: '',
        },
      ],
      tested: {},
    },
  };

  test('match snapshot correctly', () => {
    const wrapper = shallow(<Covid {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('match snapshot correctly', () => {
    const wrapper = mount(<Covid {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
