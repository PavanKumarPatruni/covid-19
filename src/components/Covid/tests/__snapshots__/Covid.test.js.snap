// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Covid component match snapshot correctly 1`] = `
<div
  className=""
>
  <div
    className="flex flex-center flex-column"
  >
    <div
      className="sm-full-width overflow-x-auto"
    />
  </div>
</div>
`;

exports[`Covid component match snapshot correctly 2`] = `
<Covid
  getCovidStateData={[Function]}
  stateData={
    Object {
      "cases_time_series": Object {},
      "statewise": Array [
        Object {
          "active": "44351",
          "confirmed": "67700",
          "deaths": "2215",
          "deltaconfirmed": "524",
          "deltadeaths": "2",
          "deltarecovered": "160",
          "lastupdatedtime": "11/05/2020 15:41:22",
          "recovered": "21130",
          "state": "Total",
          "statecode": "TT",
          "statenotes": "",
        },
        Object {
          "active": "44351",
          "confirmed": "67700",
          "deaths": "2215",
          "deltaconfirmed": "524",
          "deltadeaths": "2",
          "deltarecovered": "160",
          "lastupdatedtime": "11/05/2020 15:41:22",
          "recovered": "21130",
          "state": "Total",
          "statecode": "AP",
          "statenotes": "",
        },
      ],
      "tested": Object {},
    }
  }
>
  <div
    className=""
  >
    <div
      className="flex flex-center flex-column"
    >
      <CovidStats
        stats={
          Object {
            "active": "44351",
            "confirmed": "67700",
            "deaths": "2215",
            "deltaconfirmed": "524",
            "deltadeaths": "2",
            "deltarecovered": "160",
            "lastupdatedtime": "11/05/2020 15:41:22",
            "recovered": "21130",
            "state": "Total",
            "statecode": "TT",
            "statenotes": "",
          }
        }
      >
        <div
          className="flex covid-stats p1"
        >
          <div
            className="flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 mx2 red-color"
          >
            <h3
              className="title mt2"
            >
              Confirmed
            </h3>
            <p
              className="delta-count mt2"
            >
              [+524]
            </p>
            <p
              className="total-count mt1"
            >
              67700
            </p>
          </div>
          <div
            className="flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 mx2 blue-color"
          >
            <h3
              className="title mt2"
            >
              Active
            </h3>
            <p
              className="delta-count mt2"
            />
            <p
              className="total-count mt1"
            >
              44351
            </p>
          </div>
          <div
            className="flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 mx2 green-color"
          >
            <h3
              className="title mt2"
            >
              Recovered
            </h3>
            <p
              className="delta-count mt2"
            >
              [+160]
            </p>
            <p
              className="total-count mt1"
            >
              21130
            </p>
          </div>
          <div
            className="flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 mx2 grey-color"
          >
            <h3
              className="title mt2"
            >
              Deceased
            </h3>
            <p
              className="delta-count mt2"
            >
              [+2]
            </p>
            <p
              className="total-count mt1"
            >
              2215
            </p>
          </div>
        </div>
      </CovidStats>
      <div
        className="sm-full-width overflow-x-auto"
      >
        <StateTable
          stateList={
            Array [
              Object {
                "active": "44351",
                "confirmed": "67700",
                "deaths": "2215",
                "deltaconfirmed": "524",
                "deltadeaths": "2",
                "deltarecovered": "160",
                "lastupdatedtime": "11/05/2020 15:41:22",
                "recovered": "21130",
                "state": "Total",
                "statecode": "TT",
                "statenotes": "",
              },
              Object {
                "active": "44351",
                "confirmed": "67700",
                "deaths": "2215",
                "deltaconfirmed": "524",
                "deltadeaths": "2",
                "deltarecovered": "160",
                "lastupdatedtime": "11/05/2020 15:41:22",
                "recovered": "21130",
                "state": "Total",
                "statecode": "AP",
                "statenotes": "",
              },
            ]
          }
        >
          <table
            className="fadeInUp sm-p0"
          >
            <thead>
              <tr>
                <th>
                  State
                </th>
                <th>
                  Confirmed
                </th>
                <th>
                  Active
                </th>
                <th>
                  Recovered
                </th>
                <th>
                  Deceased
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                key="AP"
              >
                <td>
                  Total
                </td>
                <td>
                  <div
                    className="flex flex-end flex-justify-end"
                  >
                    <div
                      className="flex flex-end"
                    >
                      <ArrowUp
                        color="#ff073a"
                        size="14"
                      >
                        <svg
                          fill="#ff073a"
                          height="14"
                          stroke="#ff073a"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth="2"
                          viewBox="0 0 24 24"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <line
                            x1="12"
                            x2="12"
                            y1="19"
                            y2="5"
                          />
                          <polyline
                            points="5 12 12 5 19 12"
                          />
                        </svg>
                      </ArrowUp>
                      <p
                        className="tag mr1 "
                      >
                        524
                      </p>
                    </div>
                    <p>
                      67700
                    </p>
                  </div>
                </td>
                <td>
                  44351
                </td>
                <td>
                  <div
                    className="flex flex-end flex-justify-end"
                  >
                    <div
                      className="flex flex-end"
                    >
                      <ArrowUp
                        color="#28a745"
                        size="14"
                      >
                        <svg
                          fill="#28a745"
                          height="14"
                          stroke="#28a745"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth="2"
                          viewBox="0 0 24 24"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <line
                            x1="12"
                            x2="12"
                            y1="19"
                            y2="5"
                          />
                          <polyline
                            points="5 12 12 5 19 12"
                          />
                        </svg>
                      </ArrowUp>
                      <p
                        className="tag mr1 tag-green"
                      >
                        160
                      </p>
                    </div>
                    <p>
                      21130
                    </p>
                  </div>
                </td>
                <td>
                  <div
                    className="flex flex-end flex-justify-end"
                  >
                    <div
                      className="flex flex-end"
                    >
                      <ArrowUp
                        color="#ff073a"
                        size="14"
                      >
                        <svg
                          fill="#ff073a"
                          height="14"
                          stroke="#ff073a"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth="2"
                          viewBox="0 0 24 24"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <line
                            x1="12"
                            x2="12"
                            y1="19"
                            y2="5"
                          />
                          <polyline
                            points="5 12 12 5 19 12"
                          />
                        </svg>
                      </ArrowUp>
                      <p
                        className="tag mr1 "
                      >
                        2
                      </p>
                    </div>
                    <p>
                      2215
                    </p>
                  </div>
                </td>
              </tr>
              <tr
                key="TT"
              >
                <td>
                  Total
                </td>
                <td>
                  <div
                    className="flex flex-end flex-justify-end"
                  >
                    <div
                      className="flex flex-end"
                    >
                      <ArrowUp
                        color="#ff073a"
                        size="14"
                      >
                        <svg
                          fill="#ff073a"
                          height="14"
                          stroke="#ff073a"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth="2"
                          viewBox="0 0 24 24"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <line
                            x1="12"
                            x2="12"
                            y1="19"
                            y2="5"
                          />
                          <polyline
                            points="5 12 12 5 19 12"
                          />
                        </svg>
                      </ArrowUp>
                      <p
                        className="tag mr1 "
                      >
                        524
                      </p>
                    </div>
                    <p>
                      67700
                    </p>
                  </div>
                </td>
                <td>
                  44351
                </td>
                <td>
                  <div
                    className="flex flex-end flex-justify-end"
                  >
                    <div
                      className="flex flex-end"
                    >
                      <ArrowUp
                        color="#28a745"
                        size="14"
                      >
                        <svg
                          fill="#28a745"
                          height="14"
                          stroke="#28a745"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth="2"
                          viewBox="0 0 24 24"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <line
                            x1="12"
                            x2="12"
                            y1="19"
                            y2="5"
                          />
                          <polyline
                            points="5 12 12 5 19 12"
                          />
                        </svg>
                      </ArrowUp>
                      <p
                        className="tag mr1 tag-green"
                      >
                        160
                      </p>
                    </div>
                    <p>
                      21130
                    </p>
                  </div>
                </td>
                <td>
                  <div
                    className="flex flex-end flex-justify-end"
                  >
                    <div
                      className="flex flex-end"
                    >
                      <ArrowUp
                        color="#ff073a"
                        size="14"
                      >
                        <svg
                          fill="#ff073a"
                          height="14"
                          stroke="#ff073a"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth="2"
                          viewBox="0 0 24 24"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <line
                            x1="12"
                            x2="12"
                            y1="19"
                            y2="5"
                          />
                          <polyline
                            points="5 12 12 5 19 12"
                          />
                        </svg>
                      </ArrowUp>
                      <p
                        className="tag mr1 "
                      >
                        2
                      </p>
                    </div>
                    <p>
                      2215
                    </p>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </StateTable>
      </div>
    </div>
  </div>
</Covid>
`;
